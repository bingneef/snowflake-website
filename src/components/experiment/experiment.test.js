import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { render, fireEvent, cleanup } from "react-testing-library";

import Experiment from "./experiment";

describe("Experiment", () => {
  it("renders without error", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Experiment id={"test"} variants={[]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("matches the snapshot", () => {
    const component = renderer.create(<Experiment id={"test"} variants={[]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("defaults to the first variant", () => {
    const { container } = render(
      <Experiment id={"test"} variants={[<div />, <span />]} />
    );
    expect(container.firstChild.nodeName).toEqual("DIV");
  });
});
