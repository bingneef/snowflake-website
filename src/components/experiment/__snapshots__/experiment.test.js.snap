// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiment matches the snapshot 1`] = `null`;
